File: 0017-rotateArray/189-rotateArray.java
class Solution {
	public void rotate(int[] nums, int k) {
		int n = nums.length;
		if (n <= 1 || (k %= n) == 0) {
			return;
		}

		int[] rotatedArray = new int[n];

		for (int i = 0; i < n; i++) {
			rotatedArray[(i + k) % n] = nums[i];
		}

		for (int i = 0; i < n; i++) {
			nums[i] = rotatedArray[i];
		}
	}
}

File: 0017-RotateArray/189-rotateArray.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
 
var rotate = function(nums, k) {
    k = k % nums.length;
   for(i=0;i<k;i++)
   {
      lastElt=nums[nums.length - 1]
       nums.unshift(lastElt) ;
       nums.pop();
  
   }
  
};